Add Initial notes for lesson 11 (#39)
ReDI-School/nrw-html-and-css-2021-spring: lesson11/index.md

---
title: Lesson 11 - CSS Units
nav_order: 11
---

# CSS Units

## Class Agenda

1. Intro to CSS Units
3. CSS Font Size
4. CSS Height & Width
5. CSS Max-width

## 1. Intro to CSS Units

- What are CSS Units used for?
- What are Absolute Length Units (cm, px, in etc.) ?

> - What are Relative Length Units (em, % etc.) ?

Reviewer: since there are a few different units in each category, could you give suggestions on which units the teacher should spend more time explaining?

I think for Absolute -> px is most used. For Relative, I think `%, em, rem`. And to show differences between them

> - Explain why px is absolute but relative to device

Author: Okay, is this better:

Explain why px is relative to device (1px is calculated differently from device to device depending on the screen resolution) but size is absolute once calculated (px size is fixed on screen e.g when a user zooms in, "px" do not scale but "em", "rem" etc. scales) ?

> - Explain why px is absolute but relative to device

Reviewer: Yea, better. But still a bit confusing. 

But I think you are trying to get at this:

`What do we mean by "absolute"? What do we mean by "relative"?`

Am I right?

> - Explain why px is absolute but relative to device

Author: Hmm, not exactly but we can take it out if it is still confusing and replace it with your suggestion.

> - Explain why px is absolute but relative to device

Reviewer: Not really. I think my suggested point is different from what you intended. Let's use the updated one you shared. Can you add a link to it?

> - Explain why px is absolute but relative to device

Author: Links: https://www.w3schools.com/cssref/css_units.asp and https://chiamakaikeanyi.dev/sizing-in-css-px-vs-em-vs-rem/ 

I actually think we should leave it out and save the students the unnecessary complexity, we can add these links to the resources section for those interested in going deeper.

> - Explain why px is absolute but relative to device

Reviewer: Ok then 👍

ETA: 40mins

## 2. CSS Font Size

- `font-size` - using some of the units

ETA: 15mins

## 3. Height & Width

- cover sizing with units
- cover sizing with non-numeric CSS values (e.g auto)

ETA: 30mins

## 4. CSS Max-Width

- What is the differnce between max-width and width?

ETA: 15mins

# Exercise Description

- Increase the font-size of the "About Me", "Education" and "Language Profiencicy" headers in your portfolio website

> - Align your profile image at the center of the page and make it 20% of the screen size (the image size should adjust on different screen-sizes)

Reviewer: looks like the teacher's note is not added

---
title: Lesson 11 - CSS Units
nav_order: 11
---

# CSS Units

## Class Agenda

> 1. Intro to CSS Units

Reviewer: ```suggestion
1. Introduction to CSS Units
```

3. CSS Font Size
4. CSS Height & Width
5. CSS Max-width

> ## 1. Intro to CSS Units

Reviewer: ```suggestion
## 1. Introduction to CSS Units
```

- What are CSS Units used for?
- What are Absolute Length Units (cm, px, in etc.) ?
- What are Relative Length Units (em, % etc.) ?
- What do we mean by "absolute"? What do we mean by "relative"?
- What are the most used units and what are the differences between them? (px vs em vs rem)

ETA: 40mins

## 2. CSS Font Size

- `font-size` - using some of the units
  - Practice sizing text size using absolute unit (px)
  - Practice sizing text size using relative unit (em or rem)

ETA: 15mins

## 3. Height & Width

- cover sizing with units
- cover sizing with non-numeric CSS values (e.g auto)

ETA: 30mins

## 4. CSS Max-Width

> - What is the differnce between max-width and width?

Reviewer: ```suggestion
- What is the difference between max-width and width?
```

ETA: 15mins

# Exercise Description

- Increase the font-size of the "About Me", "Education" and "Language Profiencicy" headers in your portfolio website
- Align your profile image at the center of the page and make it 20% of the screen size (the image size should adjust on different screen-sizes)

> - At the end of this exercise, your portfolio website should like [this](./lesson-11-checkpoint.png)

Reviewer: changing this since there's a pending thing about the checkpoint mentioned [here](https://github.com/ReDI-School/nrw-html-and-css-2021-spring/pull/38#discussion_r619667160). 
```suggestion
- At the end of this exercise, your portfolio website should like what we have in Checkpoint11 branch.
```
